LeaderboardBackend/Services/Impl/AccountConfirmationService.cs

```diff
@@ -0,0 +1,48 @@
+using LeaderboardBackend.Models.Entities;
+
+namespace LeaderboardBackend.Services;
+
+public class AccountConfirmationService : IAccountConfirmationService
+{
+    private readonly ApplicationContext _applicationContext;
+    private readonly IEmailSender _emailSender;
+
+    public AccountConfirmationService(ApplicationContext applicationContext, IEmailSender emailSender)
+    {
+        _applicationContext = applicationContext;
+        _emailSender = emailSender;
+    }
+
+    public async Task<AccountConfirmation?> GetConfirmationById(Guid id)
+    {
+        return await _applicationContext.AccountConfirmations.FindAsync(id);
+    }
+
+    public async Task<AccountConfirmation> CreateConfirmation(User user, CancellationToken token = default)
```

Comment:
This function name should be changed so that it's clear that calling it enqueues an email to be sent. You should probably also move the logic in here that checks to make sure the user is in fact confirmed before you fire this off.

Reply:
Yeah, it actually makes more sense for the logic to be in the email method.